odl/operator/default_ops.py

```diff
@@ -825,26 +825,36 @@ class ConstantOperator(Operator):
     ``ConstantOperator(vector)(x) <==> vector``
     """
 
-    def __init__(self, vector, domain=None):
-        """Initialize a new instance.
+    def __init__(self, vector, domain=None, range=None):
+        """Initialize an instance.
 
         Parameters
         ----------
         vector : `LinearSpaceVector`
             The vector constant to be returned
-
-        domain : `LinearSpace`, default : vector.space
+        domain : `LinearSpace`, optional
+            default is vector.space
             The domain of the operator.
+        range : `LinearSpace`, optional
+            default : vector.space
+            The range of the operator.
         """
-        if not isinstance(vector, LinearSpaceVector):
-            raise TypeError('`vector` {!r} not a LinearSpaceVector instance'
-                            ''.format(vector))
+        if domain is None or range is None:
+            if not isinstance(vector, LinearSpaceVector):
+                raise TypeError('If either domain or range is unspecified, '
+                                '`vector` {!r} has to be a LinearSpaceVector '
+                                'instance'.format(vector))
 
         if domain is None:
             domain = vector.space
+        if range is None:
+            range = vector.space
+        if vector not in range:
+            raise TypeError('`vector` {!r} in not in the range {}.'
+                            ''.format(vector, range))
```

Comment:
Actually, the casting variant would be better here. One of the (many) principles of Python is "be generous in what you accept and strict in what you return". That last "return" could also be "store in a class". Delegating to `element` allows way more input types.